ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -265,8 +262,9 @@ addBlockSync
      )
   => ChainDbEnv m blk
   -> BlockToAdd m blk
+  -> Fuse m
   -> m ()
-addBlockSync cdb@CDB {..} BlockToAdd { blockToAdd = b, .. } = do
+addBlockSync cdb@CDB {..} BlockToAdd { blockToAdd = b, .. } hl = do
```

Comment:
What is the rationale here for now having one single big `blowAndResetFuse`?

Reply:
Im fine either way. Joris seemed to prefer fewer lifts.

Reply:
I'm confused -- aren't there multiple `blowAndResetFuse`s here? @amesgen  did you mean why it is /not/ a big `blowAndResetFuse`?

Reply:
(comment now edited, thanks for pointing this out)

Reply:
Agreed on having only one big `blockAndResetFuse`

Reply:
FTR:

> In the status quo, with multiple `blowAndResetFuse`, I immediately start thinking about scenarios like: "Could two invocations of `addBlockSync` run in parallel and be interleaved in such a way that the fuse is not blown"